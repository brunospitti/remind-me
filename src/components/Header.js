import React from "react";
import styled from "react-emotion";

import { colors } from "../globalStyles";

// styles with emotion
const Container = styled("header")`
  border-bottom: 1px solid ${colors.primary};
  position: sticky;
  top: 0;
  z-index: 10;
  padding: 20px;
`;

const StyledH1 = styled("h1")`
  font-size: 24px;
  color: ${colors.primary};
`;

export class Header extends React.PureComponent {
  render() {
    return (
      <Container>
        <StyledH1>Remind me...</StyledH1>
      </Container>
    );
  }
}
