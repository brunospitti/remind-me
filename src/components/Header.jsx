import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";

import { colors } from "../assets/globalStyles";

import { userLogout } from "../redux/actionCreators/userLogout";

import { Button } from "./basics/Button";

class Header extends React.PureComponent {
  render() {
    return (
      <StyledContainer>
        <StyledH1>To do list</StyledH1>
        <StyledButton
          clickBehavior={() => this.props.userLogout()}
          text={"Sign out"}
        />
      </StyledContainer>
    );
  }
}

// styled components
const StyledContainer = styled("header")`
  background: ${colors.light};
  border-bottom: 1px solid ${colors.primary};
  position: sticky;
  top: 0;
  z-index: 10;
  padding: 20px;
`;

const StyledH1 = styled("h1")`
  font-size: 24px;
  display: inline-block;
  color: ${colors.primary};
`;

const StyledButton = styled(Button)`
  float: right;
  margin: -5px 0 0;
  padding: 0 15px 5px;
  font-size: 1.3em;
  font-weight: 600;
  color: ${colors.complementary};
  background-color: transparent;
  border-bottom: 1px solid;
  border-radius: 0;
`;

function mapStateToProps({ user }) {
  return { user };
}

export default connect(
  mapStateToProps,
  { userLogout }
)(Header);
