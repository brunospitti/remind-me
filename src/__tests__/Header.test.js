import React from "react";
import { create } from "react-test-renderer";
import Header from "../components/Header";

test("snapshot", () => {
  const c = create(<Header />);
  expect(c.toJSON()).toMatchSnapshot();
});
