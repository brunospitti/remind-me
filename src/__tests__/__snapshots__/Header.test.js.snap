// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<header
  className="Header__Container-sc-1xm1d2j-0 kGxEuW"
>
  <h1
    className="Header__StyledH1-sc-1xm1d2j-1 iQHtNW"
  >
    To do list
  </h1>
</header>
`;
